src/transforms/prune.jl

```diff
@@ -0,0 +1,68 @@
+export prune
+
+"""
+    prune(m, xs...; simplify = true)
```

Comment:
@millerjoey Coming back to this, I had to think a bit to remember what the `simplify` switch changes. Ideally it would be obvious from the kwarg name. I think maybe something referencing `args` could be clearer. Maybe `trim_args`, or something like that?

Reply:
`trim_args` sounds good to me! I updated it.

Reply:
Looks good, thanks @millerjoey !